### ISSUE TITLE
Bundling produces corrupt .d.ts files

### ISSUE THREAD

**Issue by thekevinbrown** [wessberg/rollup-plugin-ts#151]
- Version: 1.4.2
- Rollup Version: 2.56.3
- Operating System and version (if applicable): Mac OS Big Sur v11.5.1
- Node Version (if applicable): 14.17.5
- Does it work with `tsc` (if applicable): Yes, tsc is able to build the code, but does not bundle the `.d.ts` file.

### Reproduction

Trying to build the `react-google-maps-api` package here: https://github.com/JustFly1984/react-google-maps-api/tree/37bb2ac02d6c2f03020d1d2e301a095d141e618c

### Expected Behavior

Expected successful emit of a valid `index.d.ts` file.

### Actual Behavior

Emitted file is [this one](https://unpkg.com/@react-google-maps/api@2.3.0/dist/index.d.ts), which is corrupt:

(Around line 124 in that file):
```typescript
declare class GoogleMap extends React.PureComponent<GoogleMapProps, GoogleMapState> {
    state: GoogleMapState;
    registeredEvents: google.maps.MapsEventListener[];
    mapRef: Element | null;
    getInstance: () => google.maps.Map | null;
    panTo: (latLng: google.maps.LatLng | google.maps.LatLngLiteral) => void;
    setMapCallback: () => void;
    componentDidMount(): void;
    componentDidUpdate(prevProps: GoogleMapProps): void;
    componentWillUnmount(): void;
    getRef: (ref: HTMLDivElement | null) => void;
    render(): React.ReactNode;
}
type Libraries = (leMap'

e | ort { defau | as LoadScript, | oadScript | ops } from './Lo)[];
interface LoadScriptUrlOptions {
    googleMapsApiKey: string | "";
    googleMapsClientId?: string;
    version?: string;
    language?: string;
    region?: string;
    libraries?: Libraries;
    channel?: string;
    mapIds?: string[];
}
```

Not sure what causes this. I did isolate that it's not the `hook` option causing simultaneous writes over the top of the `.d.ts` file to collide, as when I remove that configuration we get 6 `.d.ts` files, all of which have this `e | ort { defau | as LoadScript, | oadScript | ops } from './Lo)[];` line.

Issue raised in `@react-google-maps/api`: https://github.com/JustFly1984/react-google-maps-api/issues/2861

**Comment by wessberg** [comment]
Thanks for reporting. I have a feeling I know what's going on here. It is probably a node that didn't have it's TextSpan reset after being included in another SourceFile. Could also be what's causing  https://github.com/wessberg/rollup-plugin-ts/issues/145. I'll investigate.

**Comment by wessberg** [comment]
I've managed to isolate minimal reproduction, and it looks to me like a regression. I'll see if I can get it fixed right away.

**Comment by wessberg** [comment]
Hey there. Happy to report that this was an easy fix. My hunch was correct, and it was related to a node in the AST not being cloned correctly, which caused the internal TypeScript printer to print ArrayTypeNodes incorrectly. The fix has been released as part of [v1.4.3](https://github.com/wessberg/rollup-plugin-ts/releases/tag/v1.4.3)
